shared/tracker/header.render.desktop.js

```diff
@@ -2,33 +2,113 @@
 
 import React, {Component} from 'react'
 import {Header} from '../common-adapters'
+import {Icon, Text} from '../common-adapters/index'
+import {globalStyles, globalColors} from '../styles/style-guide'
+import flags from '../util/feature-flags'
 
-import type {Styled} from '../styles/common'
 import type {HeaderProps} from './header.render'
 
 export default class HeaderRender extends Component {
-  props: HeaderProps & Styled;
+  props: HeaderProps;
+  state: {showCloseWarning: boolean};
 
-  render () {
+  constructor (props: HeaderProps) {
+    super(props)
+    this.state = {showCloseWarning: false}
+  }
+
+  render (): ReactElement {
+    if (flags.tracker2) {
+      return this.render2(styles2)
+    }
+    return this.renderDefault(styles1)
+  }
+
+  renderDefault (styles: Object): ReactElement {
     return (
       <Header
-        style={{...this.props.style, ...styles.header}}
+        style={styles.header}
         icon
         title={this.props.reason}
         onClose={this.props.onClose}
       />
     )
   }
+
+  render2 (styles: Object) {
+    let warningOnClose = 'You will see this window everytime you access this folder.'
+    return (
+      <div style={styles.outer}>
+        <div style={this.state.showCloseWarning ? styles.headerCloseWarning : styles.headerNormal}>
+          <Text type='Body' style={{...styles.text, flex: 1}}>{this.state.showCloseWarning ? warningOnClose : this.props.reason}</Text>
+          <Icon type='fa-times' opacity style={styles.close}
+            onClick={() => this.props.onClose()}
+            onMouseEnter={() => this.closeMouseEnter()}
+            onMouseLeave={() => this.closeMouseLeave()} />
+        </div>
+      </div>
+    )
+  }
+
+  closeMouseEnter () {
+    this.setState({showCloseWarning: true})
+  }
+
+  closeMouseLeave () {
+    this.setState({showCloseWarning: false})
+  }
 }
 
 HeaderRender.propTypes = {
   reason: React.PropTypes.string,
-  onClose: React.PropTypes.func.isRequired,
-  style: React.PropTypes.object.isRequired
+  onClose: React.PropTypes.func.isRequired
 }
 
-const styles = {
+const styles1 = {
   header: {
     paddingLeft: 15
   }
 }
+
+const commonStyles2 = {
+  headerContainer: {
+    position: 'absolute',
+    top: 0,
+    ...globalStyles.flexBoxRow,
+    height: 90,
+    width: 320
+  }
+}
+
+const styles2 = {
+  outer: {
+    position: 'relative'
+  },
+  headerNormal: {
+    ...commonStyles2.headerContainer,
+    backgroundColor: globalColors.blue
+  },
+  headerCloseWarning: {
+    ...commonStyles2.headerContainer,
+    backgroundColor: globalColors.lowRiskWarning
+  },
```

Comment:
It would be a little nicer to do just the color here and have another style for `header`.

Then when you use it above you would just do
`style={{...styles.header, this.state.showCloseWarning ? styles.headerWarning : styles.headerNormal}}`

Then you don't need that `commonStyles2`